integration_tests/src/main/scala/com/nvidia/spark/rapids/tests/common/BenchUtils.scala

```diff
@@ -0,0 +1,399 @@
+        environment,
+        testConfiguration,
+        "csv",
+        w.writeOptions,
+        queryDescription,
+        queryPlan,
+        queryTimes)
+
+      case w: WriteParquet => BenchmarkReport(
+        filename,
+        queryStartTime.toEpochMilli,
+        environment,
+        testConfiguration,
+        "parquet",
+        w.writeOptions,
+        queryDescription,
+        queryPlan,
+        queryTimes)
+    }
+
+    writeReport(report, filename)
+  }
+
+  def readReport(file: File): BenchmarkReport = {
+    implicit val formats = DefaultFormats
+    val json = parse(file)
+    json.extract[BenchmarkReport]
+  }
+
+  def writeReport(report: BenchmarkReport, filename: String): Unit = {
+    implicit val formats = DefaultFormats
+    val os = new FileOutputStream(filename)
+    os.write(writePretty(report).getBytes)
+    os.close()
+  }
+
+  def getSparkVersion: String = {
+    // hack for databricks, try to find something more reliable?
+    if (SPARK_BUILD_USER.equals("Databricks")) {
+      SPARK_VERSION + "-databricks"
+    } else {
+      SPARK_VERSION
+    }
+  }
+
+  /**
+   * Perform a diff of the results collected from two DataFrames, allowing for differences in
+   * precision.
+   *
+   * This is only suitable for data sets that can fit in the driver's memory.
+   */
+  def compareResults(df1: DataFrame, df2: DataFrame, epsilon: Double = 0.00001): Unit = {
+
+    val result1: Seq[Seq[Any]] = collectAndSort(df1)
+    val result2: Seq[Seq[Any]] = collectAndSort(df2)
+
+    val allMatch = result1.length == result2.length && result1.zip(result2).forall {
+      case (l, r) => rowEqual(l, r, epsilon)
+    }
+
+    if (allMatch) {
+      println("Results match (allowing for differences in precision).")
+    } else {
+      val filename = s"diff-${System.currentTimeMillis()}.txt"
+      println(s"Results do not match. Writing diff to $filename")
+
+      val w = new BufferedWriter(new FileWriter(filename))
+
+      val deltas = DiffUtils.diff(
+        result1.map(_.mkString("\t")).asJava,
+        result2.map(_.mkString("\t")).asJava,
+        false).getDeltas.asScala
+
+      deltas.foreach { delta =>
+        w.write(delta.getType.toString + "\n")
+        w.write(delta.getSource.toString + "\n")
+        w.write(delta.getTarget.toString + "\n")
+      }
+
+      w.close()
+    }
+
+  }
+
+  private def collectAndSort(df: DataFrame): Seq[Seq[Any]] = {
+    // note that CPU and GPU results could, at least theoretically, end up being sorted
+    // differently because precision of floating-point values can vary but this seems to work
+    // well enough so far when comparing output from the TPC-* queries
+    df.collect().map(_.toSeq).sortBy(_.mkString(","))
+  }
+
+  private def rowEqual(row1: Seq[Any], row2: Seq[Any], epsilon: Double): Boolean = {
+    row1.zip(row2).forall {
+      case (l, r) => compare(l, r, epsilon)
+    }
+  }
+
+  // this is copied from SparkQueryCompareTestSuite
```

Comment:
It would be nice to not have to copy and paste this code here but would mean either adding a dependency on a test-jar from the tests module, or refactoring the test module to move this code under `src/main`.

Reply:
if this is a minor change, I'd vote to get this one in for this PR.

Reply:
On the other hand, it is nice that we can currently use the integration test jar with any release of the plugin (at least it works with 0.1 and 0.2 so far) so we might want to avoid adding dependencies on the plugin?

Reply:
One of my original goals with it was to have it only depend on spark, so we could run it on pure spark without anything from the plugin on the classpath.